### PR TITLE
am: Implement CreateStorage, PushInData, etc.

### PR DESCRIPTION
**Pull request by bunnei**
Basic RE work done by @ogniK5377 is the basis for this implementation. These were tested with SMO, which now gets to drawing code with these changes.

### CODE REVIEW

**Comment by greggameplayer** [comment]
@bunnei also seem to permit Rayman Legends Definitive Edition to go further

**Comment by greggameplayer** [comment]
@bunnei for me Super Mario Odyssey work the same as before

File: src/core/hle/service/am/am.cpp [hunk 1]
```diff
@@ -412,72 +505,17 @@ void ILibraryAppletCreator::CreateLibraryApplet(Kernel::HLERequestContext& ctx)
...
+void ILibraryAppletCreator::CreateStorage(Kernel::HLERequestContext& ctx) {
+    IPC::RequestParser rp{ctx};
+    const u64 size{rp.Pop<u64>()};
+    std::vector<u8> buffer(size);
 
-        rb.Push(RESULT_SUCCESS);
-        rb.PushIpcInterface<AM::IStorageAccessor>(buffer);
+    IPC::ResponseBuilder rb{rp.MakeBuilder(2, 0, 1)};
+    rb.Push(RESULT_SUCCESS);
+    rb.PushIpcInterface<AM::IStorage>(buffer);
```
Comment (Subv) [inline src/core/hle/service/am/am.cpp]: std::move

File: src/core/hle/service/am/am.cpp [hunk 2]
```diff
@@ -348,6 +349,87 @@ void ICommonStateGetter::GetPerformanceMode(Kernel::HLERequestContext& ctx) {
...
+    }
+
+private:
+    std::vector<u8> buffer;
+
+    void Open(Kernel::HLERequestContext& ctx) {
+        IPC::ResponseBuilder rb{ctx, 2, 0, 1};
+
+        rb.Push(RESULT_SUCCESS);
+        rb.PushIpcInterface<AM::IStorageAccessor>(buffer);
```
Comment (Subv) [inline src/core/hle/service/am/am.cpp]: Each call to this function copies the buffer?

Comment (bunnei) [reply src/core/hle/service/am/am.cpp]: For now, yes - until we understand better how this works. Note, the behavior of this was not changed with this PR.

**Pull request merged**